### ISSUE TITLE
Piwik tracking isn't working, consider disabling it for now?

### ISSUE THREAD

**Issue by DeeDeeG** [RefugeRestrooms/refugerestrooms#406]
Hi all.

As I mentioned in #395, Piwik tracking isn't actually working right now. Users' browsers request the piwik.js file from enspiral.info, which times out after taking a really long time trying to download the file. ~~I think the file is gone from enspiral's servers.~~ [Edit: not gone, just HTTP-only (not available over HTTPS) ]

We should probably either disable/remove Piwik from our site, or ~~do the work of setting up another Piwik instance somewhere.~~ [Edit: see comments below.]

**Comment by DeeDeeG** [comment]
Actually, the javascript file is still there. You can test it's still there by visiting:

https://analytics.enspiral.info/piwik.js

Not sure what's going on at this point, but I'm trying to find out (doing some research).

**Comment by DeeDeeG** [comment]
If we still want Piwik tracking, it would be super awesome if Enspiral would be willing to set up their servers to serve piwik.js over HTTPS. That would solve our being unable to download the file. 

It would also be better than our having to try to fetch it over HTTP [non-secure]. I'm not even sure whether that would be allowed by most browsers or not. But that would be another option to look into.

So to sum up, our options are:

1. Ask Enspiral to serve piwik.js over HTTPS
2. Fetch piwik.js over HTTP (if possible)
3. Disable Piwik tracking for now
3.1 Abandon using piwik entirely (?)
4. Leave it alone (do nothing)

**Comment by jodylecompte** [comment]
I'm not familar with Piwik, but glancing at their website it appears there is also the option of downloading and running a self hosted copy.
https://piwik.org/docs/installation/

**Comment by stardust66** [comment]
That would be a good option, but I don't think we can make it work with Heroku.

**Comment by DeeDeeG** [comment]
Maybe we could get another Heroku instance that acts as the Piwik server.

**Comment by jodylecompte** [comment]
It looks like there is a pre-packed installer for it utilizing Docker for Heroku, available with a one click install. Not sure if the cost is feasible as I am not familar with y'alls organization but looks to be pretty painless going that route (and LAMP is a tad painful compared to other stacks with Heroku). 

https://github.com/dz0ny/heroku-docker-piwik

**Comment by DeeDeeG** [comment]
Seems like everything in there is potentially free except for JawsDB MySQL (free tier only gives you a 5mb database, larger tiers are $10/month and up) 
https://elements.heroku.com/addons/jawsdb#pricing
https://elements.heroku.com/addons/jawsdb-maria#pricing

Also if we needed a paid Heroku instance for performance reasons, or if the "free dyno hours" get used up, we'd be paying monthly for that as well. Hopefully we wouldn't need to, though. ($7/month and up for paid "dynos")
https://www.heroku.com/pricing

**Comment by mi-wood** [comment]
I am ok with disabling this for now, since A) It's not working and B) I don't think anyone looks at it. I think we should just set a separate ticket for adding analytics (whether that is piwik or not).

@tkwidmer any thoughts?

**Comment by mi-wood** [comment]
https://github.com/RefugeRestrooms/refugerestrooms/compare/remove-piwik?expand=1 should we decide to go through with this

**Comment by mi-wood** [comment]
I signed up for piwik's hosted service, which has a 30 day trial and is $7 after. 

I just had to change the js to use https: and it should be good to go. 

PR is here: https://github.com/RefugeRestrooms/refugerestrooms/pull/440

**Comment by DeeDeeG** [comment]
Our free trial will end at roughly February 28th.

Any thoughts as to whether to renew the subscription when the free trial ends?

**Comment by jodylecompte** [comment]
Digital marketing and analytics is a large majority of my work, so I definitely understand getting analytics how you want them is difficult and having to re-do things sucks. So myself and another GitHubber (@jdorman0) collected $40 dollars to go towards paying up for a few months, or if y'all end up deciding Piwik doesn't provide enough value to warrant paying for then maybe it'll cover a few months of hosting or getting the core maintainers a cup of coffee.

What is the best way to send this? I checked out the Patreon but didn't see an option for a one-time donation.

**Comment by mi-wood** [comment]
@jodylecompte Sorry for the late reply, but thanks so much! That will definitely help with subscription costs for a while!
